### PR TITLE
GLFW migration

### PR DESCRIPTION
**Pull request by GiantLuigi4**
All events now expect GLFW mappings instead of AWT mappings

### CODE REVIEW

File: java/org/cef/event/CefMouseEvent.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public static final int BUTTON1_MASK = 0x10;
+    public static final int BUTTON2_MASK = 0x20;
+    public static final int BUTTON3_MASK = 0x40;
+
+    public int id;
+    public int x;
+    public int y;
+    public int modifs;
+    public int clickCount;
+    public int button;
```
Comment (ds58) [inline java/org/cef/event/CefMouseEvent.java:17]: Make private; we have getters

File: java/org/cef/event/CefMouseEvent.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    public static final int MOUSE_EXIT = 505;
+    /* modifier constants */
+    public static final int BUTTON1_MASK = 0x10;
+    public static final int BUTTON2_MASK = 0x20;
+    public static final int BUTTON3_MASK = 0x40;
+
+    public int id;
+    public int x;
+    public int y;
+    public int modifs;
```
Comment (ds58) [inline java/org/cef/event/CefMouseEvent.java]: `modifiers`

File: java/org/cef/event/CefMouseEvent.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    public int getX() {
+        return x;
+    }
+
+    public int getY() {
+        return y;
+    }
+
+    public int getModifiersEx() {
```
Comment (ds58) [inline java/org/cef/event/CefMouseEvent.java]: `getModifiers()`

File: java/org/cef/event/CefMouseWheelEvent.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
+package org.cef.event;
+
+public class CefMouseWheelEvent {
+    /* id constants */
+    public static final int WHEEL_UNIT_SCROLL = 0;
+    public static final int WHEEL_BLOCK_SCROLL = 1;
+
+    int id, delta, x, y, modifiers;
```
Comment (ds58) [inline java/org/cef/event/CefMouseWheelEvent.java]: Make public, put each on a new line

File: java/org/cef/event/CefMouseWheelEvent.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    int amount = 32; // TODO: might want to make this change dependent on system?
```
Comment (ds58) [inline java/org/cef/event/CefMouseWheelEvent.java]: Make public

File: java/org/cef/event/CefMouseWheelEvent.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+    public int getX() {
+        return x;
+    }
+
+    public int getY() {
+        return y;
+    }
+
+    public int getModifiersEx() {
```
Comment (ds58) [inline java/org/cef/event/CefMouseWheelEvent.java]: `getModifiers()`

File: java/org/cef/handler/CefResourceHandlerAdapter.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 package org.cef.handler;
 
 import org.cef.callback.CefCallback;
+import org.cef.misc.BoolRef;
```
Comment (ds58) [inline java/org/cef/handler/CefResourceHandlerAdapter.java]: unused import?

File: native/CefBrowser_N.cpp [hunk 8]
```diff
@@ -1902,11 +1926,11 @@ Java_org_cef_browser_CefBrowser_1N_N_1SendMouseWheelEvent(
   cef_event.x = x;
   cef_event.y = y;
 
-  cef_event.modifiers = GetCefModifiers(env, cls, modifiers);
+  cef_event.modifiers = GetCefModifiersGlfw(env, cls, modifiers);
 
-  if (scroll_type == JNI_STATIC(WHEEL_UNIT_SCROLL)) {
+  if (scroll_type == 0) {
```
Comment (ds58) [inline native/CefBrowser_N.cpp]: Add a comment saying this means WHEEL_UNIT_SCROLL

File: native/CefBrowser_N.cpp [hunk 9]
```diff
@@ -1880,20 +1903,21 @@ Java_org_cef_browser_CefBrowser_1N_N_1SendMouseWheelEvent(
...
                         &scroll_type) ||
-      !CallJNIMethodI_V(env, cls, mouse_wheel_event, "getWheelRotation",
+      !CallJNIMethodI_V(env, objClass, mouse_wheel_event, "getWheelRotation",
                         &delta) ||
-      !CallJNIMethodI_V(env, cls, mouse_wheel_event, "getX", &x) ||
-      !CallJNIMethodI_V(env, cls, mouse_wheel_event, "getY", &y) ||
-      !CallJNIMethodI_V(env, cls, mouse_wheel_event, "getModifiersEx",
+      !CallJNIMethodI_V(env, objClass, mouse_wheel_event, "getX", &x) ||
+      !CallJNIMethodI_V(env, objClass, mouse_wheel_event, "getY", &y) ||
+      !CallJNIMethodI_V(env, objClass, mouse_wheel_event, "getModifiersEx",
```
Comment (ds58) [inline native/CefBrowser_N.cpp]: change to `getModifiers`

File: native/CefBrowser_N.cpp [hunk 10]
```diff
@@ -1815,62 +1841,59 @@ Java_org_cef_browser_CefBrowser_1N_N_1SendMouseEvent(JNIEnv* env,
...
 
   int event_type, x, y, modifiers;
-  if (!CallJNIMethodI_V(env, cls, mouse_event, "getID", &event_type) ||
-      !CallJNIMethodI_V(env, cls, mouse_event, "getX", &x) ||
-      !CallJNIMethodI_V(env, cls, mouse_event, "getY", &y) ||
-      !CallJNIMethodI_V(env, cls, mouse_event, "getModifiersEx", &modifiers)) {
+  if (!CallJNIMethodI_V(env, objClass, mouse_event, "getID", &event_type) ||
+      !CallJNIMethodI_V(env, objClass, mouse_event, "getX", &x) ||
+      !CallJNIMethodI_V(env, objClass, mouse_event, "getY", &y) ||
+      !CallJNIMethodI_V(env, objClass, mouse_event, "getModifiersEx", &modifiers)) {
```
Comment (ds58) [inline native/CefBrowser_N.cpp]: change to `getModifiers`

File: native/CefBrowser_N.cpp [hunk 11]
```diff
@@ -1600,66 +1626,66 @@ Java_org_cef_browser_CefBrowser_1N_N_1SendKeyEvent(JNIEnv* env,
...
+  JNI_STATIC_DEFINE_INT(env, cls, GLFW_REPEAT);
 
   int event_type, modifiers;
   char16 key_char;
-  if (!CallJNIMethodI_V(env, cls, key_event, "getID", &event_type) ||
-      !CallJNIMethodC_V(env, cls, key_event, "getKeyChar", &key_char) ||
-      !CallJNIMethodI_V(env, cls, key_event, "getModifiersEx", &modifiers)) {
+  if (!CallJNIMethodI_V(env, objClass, key_event, "getID", &event_type) ||
+      !CallJNIMethodC_V(env, objClass, key_event, "getKeyChar", &key_char) ||
+      !CallJNIMethodI_V(env, objClass, key_event, "getModifiersEx", &modifiers)) {
```
Comment (ds58) [inline native/CefBrowser_N.cpp]: change to `getModifiers`

File: java/org/cef/event/CefKeyEvent.java [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+
+public class CefKeyEvent {
+    /* id constants */
+    public static final int KEY_PRESS = 1;
+    public static final int KEY_RELEASE = 0;
+    public static final int KEY_TYPE = 2;
+
+    public int keyCode;
+    public int id;
+    public int modifiersEx;
```
Comment (ds58) [inline java/org/cef/event/CefKeyEvent.java]: change to `modifiers`

File: java/org/cef/event/CefKeyEvent.java [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    public char keyChar;
+    public long scancode;
```
Comment (ds58) [inline java/org/cef/event/CefKeyEvent.java:13]: make private; we have getters

File: java/org/cef/event/CefKeyEvent.java [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+        this.id = id;
+        this.modifiersEx = modifiersEx;
+        this.keyChar = keyChar;
+    }
+
+    public int getID() {
+        return id;
+    }
+
+    public int getModifiersEx() {
```
Comment (ds58) [inline java/org/cef/event/CefKeyEvent.java]: change to `getModifiers()`

**Review by ds58 — COMMENTED** [review]
Just some naming and formatting things

File: java/org/cef/handler/CefResourceHandlerAdapter.java (hunk 7 above)
Comment (GiantLuigi4) [reply java/org/cef/handler/CefResourceHandlerAdapter.java]: Leftover from a different thing I did, yeah

File: native/CefBrowser_N.cpp (hunk 8 above)
Comment (GiantLuigi4) [reply native/CefBrowser_N.cpp]: Good point

**Pull request merged**